ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java

```diff
@@ -1171,9 +1218,13 @@ private void setResourceConfig(String dcName, String resourceName, ConfigAccesso
     }
     partitionCapacityMap.put(DEFAULT_PARTITION_KEY, Collections.singletonMap(DISK_KEY,
         clusterConfigFields.partitionDiskWeightInGB + PARTITION_BUFFER_CAPACITY_FOR_INDEX_FILES_IN_GB));
-
-    // 3. Update resource configs in helix
     resourceConfig.setPartitionCapacityMap(partitionCapacityMap);
+
+    // 3. Set semi-auto placement as preference list so that no replicas are moved.
+    Map<String, List<String>> semiAutoPreferenceLists = idealState.getPreferenceLists();
+    resourceConfig.setPreferenceLists(semiAutoPreferenceLists);
```

Comment:
can we have another parameter to control this?

Reply:
Done